srcs/builtin_fts/export.c

```diff
@@ -2,24 +2,80 @@
 
 /*
  * Error handling
- * Export without options - thus it will ignore any flags
- * and not throw out an error
+ * Export without options - thus it will ignore any flags and not throw out an error
+ * Starts with digit or with = : bash: export: `1test': not a valid identifier
+ * Has a symbol (/ % * + , - . : ? [] {} ^ ~): bash: export: `/test': not a valid identifier
+ * Some symbols are ignored or lead to special behaviour (# & $ ! ; and parsing stuff)
+ * Some symbols are accepted (_)
+ *
+ * Each argument leads to a new variable
+ * The variable only has a value if there's an =
+ * If one prompts an error, the next one is still added
  */
 
+static void	print_export(t_list *env, int fd_in)
+{
+	while (env)
+	{
+		ft_putstr_fd("declare -x ", fd_in);
+		ft_putstr_fd(get_env_lst_name(env), fd_in);
+		if (get_env_lst_value(env))
+		{
+			write(fd_in, "=\"", 2);
+			ft_putstr_fd(get_env_lst_value(env), fd_in);
+			write(fd_in, "\"", 1);
+		}
+		write(fd_in, "\n", 1);
+		env = env->next;
+	}
+}
+
+static int	add_env_list(char *arg, t_list *env)
+{
+	int		i;
+	t_env	*env_var;
+	
+	if (!ft_isalpha(arg[0]) && arg[0] != '_')
+		return (1);
+	i = 1;
+	while (arg[i] && (ft_isalnum(arg[i]) || arg[i] == '_'))
+		i++;
+	if (arg[i] && arg[i] != '=')
+		return (1);
+	env_var = (t_env *)malloc(sizeof(t_env));
```

Comment:
why not calloc or exit ?

Reply:
😂 very fair point. because I forgot. does exit free all memory still lying around?

Reply:
Done :)